src/plugins/safeSnap/utils/umaModule.ts

```diff
@@ -481,20 +469,11 @@ export const getModuleDetailsUmaGql = async (
       }
     : undefined;
 
-  const thisProposalTransactionsProposedEvents =
-    transactionsProposedEvents.filter(
-      event => toUtf8String(event.args?.explanation) === explanation
-    );
-
-  const assertion = thisProposalTransactionsProposedEvents.map(
-    tx => tx.args?.assertionId
-  );
-
-  const assertionIds = executionEvents.map(tx => tx.args?.assertionId);
-
-  const proposalExecuted = assertion.some(assertionId =>
-    assertionIds.includes(assertionId)
-  );
+  const proposals = await findProposalsGql(network, {
+    assertionId,
+    explanation
+  });
+  const proposalExecuted = proposals?.[0]?.executed ? true : false;
```

Comment:
this is the only data we needed from the events, if the proposal was executed or not